polkadot-parachain/src/command.rs

```diff
@@ -40,55 +40,54 @@ use sp_core::hexdisplay::HexDisplay;
 use sp_runtime::traits::{AccountIdConversion, Block as BlockT};
 use std::{io::Write, net::SocketAddr};
 
-trait IdentifyChain {
-	fn is_shell(&self) -> bool;
-	fn is_seedling(&self) -> bool;
-	fn is_statemint(&self) -> bool;
-	fn is_statemine(&self) -> bool;
-	fn is_westmint(&self) -> bool;
-	fn is_contracts_rococo(&self) -> bool;
+enum Runtime {
+	Shell,
+	Seedling,
+	Statemint,
+	Statemine,
+	Westmint,
+	ContractsRococo,
```

Comment:
Maybe there should be a `Generic` enum arm here and if so we can remove the need to wrap this in an option.

Reply:
Yeah, that's a tad nicer I think.